detections/cloud/azure_ad_device_code_authentication.yml

```diff
@@ -0,0 +1,74 @@
+name: Azure AD Device Code Authentication
+id: d68d8732-6f7e-4ee5-a6eb-737f2b990b91
+version: 1
+date: '2023-08-03'
+author: Mauricio Velazco, Gowthamaraj Rajendran,  Splunk
+status: production
+type: TTP
+data_source: []
+description: The following analytic identifies the execution of the Azure Device Code Phishing attack, 
+  which can lead to Azure Account Take-Over (ATO). The detection leverages Azure AD logs specifically 
+  focusing on authentication requests to identify the attack. This technique involves creating malicious 
+  infrastructure, bypassing Multi-Factor Authentication (MFA), and bypassing Conditional Access Policies (CAPs). 
+  The attack aims to compromise users by sending them phishing emails from attacker-controlled domains and trick 
+  the victims into performing OAuth 2.0 device authentication. A successful execution of this attack can result 
+  in adversaries gaining unauthorized access to Azure AD, Exchange mailboxes, and the target's Outlook Web Application (OWA). 
+  This attack technique was detailed by security researchers including Bobby Cooke, Stephan Borosh, and others. 
+  It's crucial for organizations to be aware of this threat, as it can lead to unauthorized access and potential data breaches.
+search: '`azuread` category=SignInLogs "properties.authenticationProtocol"=deviceCode 
+  | rename properties.* as * 
+  | stats values(userPrincipalName) by _time, ipAddress, appDisplayName, userAgent
```

Comment:
add : values(userPrincipalName) as user 

lets use CIM field names. when you look at this data in splunk - do you see src_ip or src instead of ipAddress. 
You can identify field names which are CIM specific usually in lowercase format

Reply:
updated to use CIM fields. now that I'm using the new sourcetype that maps to CIM, using CIM fields.